lib/resources/entities.js

```diff
@@ -95,17 +95,19 @@ module.exports = (service, endpoint) => {
 
   }));
 
-  service.get('/projects/:id/datasets/:name/entities/:uuid/diffs', endpoint(async ({ Projects }, { params, queryOptions }) => {
+  service.get('/projects/:projectId/datasets/:name/entities/:uuid/diffs', endpoint(async ({ Datasets, Entities }, { params, auth, queryOptions }) => {
 
-    await Projects.getById(params.id, queryOptions).then(getOrNotFound);
+    const dataset = await Datasets.get(params.projectId, params.name, true).then(getOrNotFound);
+
+    await auth.canOrReject('entity.read', dataset);
+
+    const defs = await Entities.getAllDefs(dataset.id, params.uuid, queryOptions);
+
+    // it means there's no entity with the provided UUID
+    if (defs.length === 0) return reject(Problem.user.notFound());
+
+    return diffEntityData(defs.map(d => d.data));
```

Comment:
I think it will not pick up changes to the label because the label is not part of data.